tests/codegen/fcode/scripts/openmp.py

```diff
@@ -0,0 +1,32 @@
+# pylint: disable=missing-function-docstring, missing-module-docstring/
+
+from numpy import zeros
+
+n = 800
+m = 1600
+p = 800
+
+a = zeros((n,m))
+b = zeros((m,p))
+c = zeros((n,p))
+
+#$ omp parallel
+#$ omp do schedule(runtime)
+for i in range(0, n):
+    for j in range(0, m):
+        a[i,j] = i-j
+#$ omp end do nowait
+
+#$ omp do schedule(runtime)
+for i in range(0, m):
+    for j in range(0, p):
+        b[i,j] = i+j
+#$ omp end do nowait
+
+#$ omp do schedule(runtime)
+for i in range(0, n):
+    for j in range(0, p):
+        for k in range(0, p):
+            c[i,j] = c[i,j] + a[i,k]*b[k,j]
+#$ omp end do
+#$ omp end parallel
```

Comment:
Why do you feel this file is necessary? Does `schedule(runtime)` not compile yet?

The files in codegen/ are only tested to see if a file can be generated. They are therefore not very robust tests as they don't test if the file compiles or if it produces the expected result. They mostly exist to prevent regression on syntax that is not fully implemented or for historical reasons (these were once the only tests).  This file obviously does not fall into the latter category. Can you explain what is not fully implemented (presumably this is something that is not implemented at all in the c printer as you have not created an equivalent `tests/codegen/ccode/scripts/openmp.py` file?)

Reply:
there is a lot of missing pragmas, but this test is just to show the difference between OMP syntax in C and Fortran.
in this example there is `nowait` clause at the end of the loop construct, pyccel will generate a code that contain the `nowait` clause at the end of the loop construct which it will work in fortran but it will fail in C because the `nowait` clause must be at the start of the loop construct in C.

Reply:
Does that mean this test should pass for Fortran but only pass the codegen stage for c?

Reply:
exactly, but because of arrays this test will raise a non implemented error for C

Reply:
Then it sounds to me like this should be an epyccel test for fortran and could currently only be a codegen test for c if arrays were working. Is that right?

Reply:
Yes, in the OpenMP epyccel tests we already have tests with arrays that work for fortran and not for C

Reply:
I agree with @EmilyBourne but I suggest we keep this test as it is. Do we have a similar test in epyccel?

Reply:
Then I am not sure I understand what this file is testing that is not tested in epyccel. I can see the utility of a `tests/codegen/ccode/scripts/openmp.py` which is marked as xfail until arrays are working. But this test is only testing fortran

Reply:
then i should remove this file because he is doing the work of tests in epyccel

Reply:
Fine